wgpu-native/src/device.rs

```diff
@@ -63,29 +63,88 @@ pub extern "C" fn wgpu_device_create_pipeline_layout(
     device_id: DeviceId,
     desc: binding_model::PipelineLayoutDescriptor,
 ) -> PipelineLayoutId {
-    unimplemented!()
+    let bind_group_layout_ids = unsafe { slice::from_raw_parts(desc.bind_group_layouts, desc.bind_group_layouts_length) };
+    // TODO: only lock mutex once?
+    let descriptor_set_layouts = bind_group_layout_ids.iter().map(|id| registry::BIND_GROUP_LAYOUT_REGISTRY.get_mut(*id)).collect::<Vec<_>>();
+    let device = &mut registry::DEVICE_REGISTRY.get_mut(device_id).device;
```

Comment:
all the device methods are supposed to be `&self`, so shouldn't need `get_mut` here